internal/testutil/types.go

```diff
@@ -31,6 +31,9 @@ func GetClientFromMyContext(ctx MyContext) (graphql.Client, error)     { return
 const dateFormat = "2006-01-02"
 
 func MarshalDate(t *time.Time) ([]byte, error) {
+	if t == nil || t.IsZero() {
+		return []byte("null"), nil
+	}
```

Comment:
Why did you add this back in?

Reply:
The `t == nil` should be unnecessary, I just put it there for completeness.  But `t.IsZero()` we do want, to avoid marshaling a `time.Time` field to `"0001-01-01"`, since genqlient matches `encoding/json` in counting zero structs as nonempty for purposes of `omitempty`.